cmd/snap/cmd_find.go

```diff
@@ -113,15 +113,24 @@ func (x *cmdFind) Execute(args []string) error {
 	}
 
 	// magic! `snap find` returns the featured snaps
-	if x.Positional.Query == "" && x.Section == "" {
+	showFeatured := (x.Positional.Query == "" && x.Section == "")
+	if showFeatured {
 		x.Section = "featured"
+		fmt.Fprintf(Stdout, i18n.G("No search term specified, showing the featured snaps:\n\n"))
 	}
 
-	return findSnaps(&client.FindOptions{
+	if err := findSnaps(&client.FindOptions{
```

Comment:
If I understand this correctly, the actual search is performed after “showing the featured snaps” has already been printed.

If so, the result might be:
`No search term specified, showing the featured snaps:`
`error: cannot list snaps: Get https://api.snapcraft.io/api/v1/snaps/search?confinement=strict%2Cclassic&fields=anon_download_url%2Carchitecture%2Cchannel%2Cdownload_sha3_384%2Csummary%2Cdescription%2Cdeltas%2Cbinary_filesize%2Cdownload_url%2Cepoch%2Cicon_url%2Clast_updated%2Cpackage_name%2Cprices%2Cpublisher%2Cratings_average%2Crevision%2Cscreenshot_urls%2Csnap_id%2Clicense%2Cbase%2Csupport_url%2Ccontact%2Ctitle%2Ccontent%2Cversion%2Corigin%2Cdeveloper_id%2Cprivate%2Cconfinement%2Cchannel_maps_list&q=&section=featured: dial tcp: lookup api.snapcraft.io on [::1]:53: read udp [::1]:38309->[::1]:53: read: connection refused`

Obviously that offline message could be improved. But no matter how polite it was, I think it would be better to delay printing “showing featured snaps” until you know for sure whether you’ll be showing any snaps at all.

Reply:
Fwiw, the offline error is being worked on in https://github.com/snapcore/snapd/pull/4435